### PR TITLE
Updated middleware to redirect user to index if signed in without referrer

### PR DESCRIPTION
**Pull request by SVarallo**

### CODE REVIEW

File: src/authentication/user/middleware/index.ts [hunk 1]
```diff
@@ -3,6 +3,15 @@ import logger from "../../../logger";
 import * as pageURLs from "../../../model/page.urls";
 
 export default (req: Request, res: Response, next: NextFunction) => {
+  const referringPageURL = req.header("Referer") as string;
+
+  logger.debug("Check if user has referrer");
+
+  if (referringPageURL === undefined) {
+
```
Comment (markpit) [inline src/authentication/user/middleware/index.ts]: don't need this blank line

File: test/authentication/user/middleware/index.spec.unit.ts [hunk 2]
```diff
@@ -28,7 +28,14 @@ describe("Authentication middleware", () => {
...
     const response = await request(app)
       .get("/promise-to-file/");
-    expect(response.status).toEqual(200);
+    expect(response.status).toEqual(302);
+  });
+
+  it("should redirect to start page if /promise-to-file/* called and signed in", async () => {
+    const response = await request(app)
+      .get("/promise-to-file/company-number")
+      .expect("Location", "/promise-to-file");
```
Comment (markpit) [inline test/authentication/user/middleware/index.spec.unit.ts]: I think you should populate the referer here, it looks liek it is redirecting for the wrong reason ie. referer is blank

File: src/authentication/user/middleware/index.ts [hunk 3]
```diff
@@ -3,29 +3,27 @@ import logger from "../../../logger";
 import * as pageURLs from "../../../model/page.urls";
 
 export default (req: Request, res: Response, next: NextFunction) => {
+  const referringPageURL = req.header("Referer") as string;
 
-  logger.debug("Check if user is authenticated");
+  logger.debug("Check if user has referrer ");
+  if (referringPageURL === undefined) {
```
Comment (mwestacott) [inline src/authentication/user/middleware/index.ts]: This new logic slightly changes the behaviour  of this story (which is 'done') - https://companieshouse.atlassian.net/browse/LFA-1215. This new way of doing things may well be fine, but worth checking with analyst and tester.

File: src/authentication/user/middleware/index.ts [hunk 4]
```diff
@@ -3,29 +3,27 @@ import logger from "../../../logger";
 import * as pageURLs from "../../../model/page.urls";
 
 export default (req: Request, res: Response, next: NextFunction) => {
+  const referringPageURL = req.header("Referer") as string;
 
-  logger.debug("Check if user is authenticated");
+  logger.debug("Check if user has referrer ");
```
Comment (mwestacott) [inline src/authentication/user/middleware/index.ts]: Blank space at end not needed.

File: src/authentication/user/middleware/index.ts (hunk 1 above)
Comment (SVarallo) [reply src/authentication/user/middleware/index.ts]: 👍

File: test/authentication/user/middleware/index.spec.unit.ts (hunk 2 above)
Comment (SVarallo) [reply test/authentication/user/middleware/index.spec.unit.ts]: Changed the way the unit tests are designed to better test the new functionality

File: src/authentication/user/middleware/index.ts (hunk 4 above)
Comment (SVarallo) [reply src/authentication/user/middleware/index.ts]: 👍 +1

File: src/authentication/user/middleware/index.ts (hunk 3 above)
Comment (SVarallo) [reply src/authentication/user/middleware/index.ts]: Have discussed with Tester and will also discuss with analyst before merging

Comment (SVarallo) [reply src/authentication/user/middleware/index.ts]: Have discussed with analyst and they are happy, merging now

**Pull request merged**